client/peerset/src/peersstate.rs

```diff
@@ -128,21 +128,21 @@ enum MembershipState {
 }
 
 impl MembershipState {
-	/// Returns `true` for `In` and `Out`.
+	/// Returns `true` for [`MembershipState::In`] and [`MembershipState::Out`].
 	fn is_connected(self) -> bool {
-		match self {
-			MembershipState::NotMember => false,
-			MembershipState::In => true,
-			MembershipState::Out => true,
-			MembershipState::NotConnected { .. } => false,
-		}
+		matches!(self, Self::In | Self::Out)
+	}
```

Comment:
I also prefer the old exhaustive code

Reply:
Given the specific function name and/or inline doc comment, the context also needs to be updated when a new variant is added. Anyway, suggestions applied :p.